webtranslate/pages/login.py

```diff
@@ -0,0 +1,71 @@
+"""
+Login/logout
+"""
+from urllib.parse import quote
+from webtranslate.bottle import route, request, redirect
+from webtranslate.protect import protected, start_session, stop_session
+from webtranslate import utils, users
+
+def login_success(r):
+    if r is None or not r.startswith("/"):
+        r = "/userprofile"
+
+    redirect(r + "?message=" + quote("Login successful!"))
+
+@route('/login', method = 'GET')
+@protected(['login', '-', '-'])
+def login(userauth):
+    req_redirect = request.query.get("redirect")
+    req_login = request.query.get('login')
+
+    if userauth.is_auth:
+        login_success(req_redirect)
+    elif users.oauth_redirect is not None:
+        session, url = users.oauth_redirect(req_redirect, req_login)
+        start_session(session)
+        redirect(url)
+    elif users.get_authentication is not None:
+        return utils.template('login', userauth = userauth, req_login = req_login, req_redirect = req_redirect)
+    else:
+        utils.redirect("/")
+
+@route('/login', method = 'POST')
+@protected(['login', '-', '-'])
+def login(userauth):
+    if users.get_authentication is None:
+        utils.redirect("/")
```

Comment:
from what I understand, this means that there is no longer defined. Personally, I am a big fan of returning 404 pages in those cases, instead of redirecting uses away. This can be very unclear from a debugging point of view. Happens more often throughout this code btw, but only mentioning it once :)